File: 6-kyu/convert-string-to-camel-case/python/solution.py
import re

def to_camel_case(text):
    if not text:
        return ''
    text_split = re.split('\-|\_', text)
    ret = text_split[0]
    for i in range(1, len(text_split)):
        ret += text_split[i][0].upper() + text_split[i][1:]
    return ret

File: 6-kyu/convert-string-to-camel-case/c++/solution.cpp
#include <cctype>

std::string to_camel_case(std::string text)
{
    std::string ret = "";
    bool first_word = true;
    std::string::size_type start = 0, end;
    for (unsigned i = 0; i < text.length(); i++) {
        if (text[i] == '-' || text [i] == '_') {
            end = i;
            std::string word = text.substr(start, end - start);
            if (first_word) first_word = false;
            else word[0] = std::toupper(word[0]);
            ret += word;
            start = end + 1;
        }
    }
    std::string word = text.substr(start);
    word[0] = std::toupper(word[0]);
    ret += word;
    return ret;
}